### PR TITLE
Refactor: move global ratingState into express-session

### PR DESCRIPTION
**Pull request by elaguerta**
This moves the ratingState module into the Session object initialized by express-session. I am proposing this because it is better practice to store state in the Session object, and if/when we do return to 2fa, it might be helpful to have ratingState stored only in the session so that we don't have to deal with the server restarting or inconsistent browser behavior.
Note on the many changes: I installed 'es-lint prettier' on vscode to format according to eslint rules... and it changed like every character. SORRY!

All existing Postman tests still pass, but I haven't done thorough manual testing. 
image
image
image

### CODE REVIEW

**Comment by elaguerta** [comment]
Re-opening with a couple of changes to pass the user's selections from the query params to the Session.ratingState.

**Comment by elaguerta** [comment]
Came back here to say I feel pretty strongly that we should move the ratingState to Session storage, but we should at this point probably hold off until A6 and A4 fixes merge in. If you either of you get a chance to review this, I'd appreciate it, but please don't merge it in yet, let's consider this one a draft.

**Comment by elaguerta** [comment]
Added some more ratingState handling and a few more tests. All Postman tests pass. 
image

File: routes/main.js
Comment (elaguerta) [inline routes/main.js:65]:
```suggestion
  router.get("/register", authHandler.isNotAuthenticated, initializeRatingState, function (req, res) {
```

**Review by seokhchoi — APPROVED** [review]
Awesome. Thanks!

**Pull request merged**
